sdk/communication/Shared/src/RedirectPolicy.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Communication.Pipeline
+{
+    internal class RedirectPolicy : HttpPipelinePolicy
+    {
+        private const string LOCATION_HEADER_STRING = "LOCATION";
+        private readonly int maxRedirects;
+
+        public RedirectPolicy(int maxRedirects = 10)
+        {
+            this.maxRedirects = maxRedirects;
+        }
+
+        public override void Process(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
+        {
+            var redirectCount = 0;
+
+            while (true)
+            {
+                ProcessNext(message, pipeline);
+
+                if (!TryGetRedirect(message, out Uri location))
+                {
+                    break;
+                }
+
+                if (++redirectCount > maxRedirects)
+                {
+                    break;
+                }
+
+                message.Request.Uri.Reset(location);
+            }
+        }
+
+        public override async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
+        {
+            var redirectCount = 0;
+
+            while (true)
+            {
+                await ProcessNextAsync(message, pipeline).ConfigureAwait(false);
+
+                if (!TryGetRedirect(message, out Uri location))
+                {
+                    break;
```

Comment:
It might be better to throw here so the client doesn't try to interpret the redirect response and fail with an obscure error.

Reply:
Wrong comment location. I mean inside the `if (++redirectCount > maxRedirects)` condition.

Reply:
This is the normal flow path, where the request was issued successfully and the redirect does not need to do anything.

Reply:
I can put a comment in to make this clear.

Reply:
Updated to make the code clearer.